File: app/src/main/java/com/tokuraku/models/Word.java
package com.tokuraku.models;

public class Word {
    private long    id;
    private String kanji;
    private String furigana;
    private String translation;
    private String ex_jp;
    private String ex_eng;

    public Word(String kanji, String furigana, String translation, String ex_jp, String ex_eng) {

        this.kanji = kanji;
        this.furigana = furigana;
        this.translation = translation;
        this.ex_jp = ex_jp;
        this.ex_eng = ex_eng;
    }
}

File: app/src/main/java/com/kamyart/tokuraku/model/Word.kt
package com.kamyart.tokuraku.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "word_table")
class Word (
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") private var id: Int,
    @ColumnInfo(name = "kanji") private var kanji: String,
    @ColumnInfo(name = "furigana") private var furigana: String,
    @ColumnInfo(name = "translation") private var translation: ArrayList<String>,
    @ColumnInfo(name = "examples") private var examples: Map<String, String>,

    ){
    //getters
    fun getId() = id
    fun getKanji() = kanji
    fun getFurigana() = furigana
    fun getTranslation() = translation
    fun getExamples() = examples

    //setters
    fun setKanji(kanji: String){this.kanji = kanji}
    fun setFurigana(furigana: String){this.furigana = furigana}
    fun setTranslation(translation: ArrayList<String>){this.translation = translation}
    fun examples(examples: Map<String, String>){this.examples = examples}
}
